cms/djangoapps/contentstore/views/tests/test_videos.py

```diff
@@ -854,6 +856,160 @@ def test_video_image_validation_message(self, image_data, error_message):
+        ),
+        # Success
+        (
+            {
+                'provider': TranscriptProvider.CIELO24,
+                'cielo24_fidelity': 'PROFESSIONAL',
+                'cielo24_turnaround': 'STANDARD',
+                'preferred_languages': ['en']
+            },
+            ''
+        ),
+        (
+            {
+                'provider': TranscriptProvider.THREE_PLAY_MEDIA,
+                'three_play_turnaround': 'default',
+                'preferred_languages': ['en']
+            },
+            ''
+        )
+    )
+    @ddt.unpack
+    def test_video_transcript(self, preferences, error_message):
+        """
+        Tests that transcript handler works correctly.
+        """
+        video_transcript_url = self.get_url_for_course_key(self.course.id)
+        response = self.client.post(
+            video_transcript_url,
+            json.dumps({
+                'provider': preferences.get('provider', ''),
+                'cielo24_fidelity': preferences.get('cielo24_fidelity', ''),
+                'cielo24_turnaround': preferences.get('cielo24_turnaround', ''),
+                'three_play_turnaround': preferences.get('three_play_turnaround', ''),
+                'preferred_languages': preferences.get('preferred_languages', []),
+            }),
+            content_type='application/json'
+        )
+        response = json.loads(response.content)
+        if error_message:
+            self.assertEqual(response['status'], 400)
+            self.assertEqual(response['error'], error_message)
+        else:
+            self.assertTrue('transcript_preferences' in response)
+
+    @ddt.data(
+        None,
+        {
+            'provider': TranscriptProvider.CIELO24,
+            'cielo24_fidelity': 'PROFESSIONAL',
+            'cielo24_turnaround': 'STANDARD',
+            'preferred_languages': ['en']
+        }
+    )
+    @override_settings(AWS_ACCESS_KEY_ID='test_key_id', AWS_SECRET_ACCESS_KEY='test_secret')
+    @patch('boto.s3.key.Key')
+    @patch('boto.s3.connection.S3Connection')
+    def test_transcript_preferences_metadata(self, transcript_preferences, mock_conn, mock_key):
+        """
+        Tests that transcript preference metadata is only set if it is transcript
+        preferences are present in request data.
+        """
+        file_name = 'test-video.mp4'
+        request_data = {'files': [{'file_name': file_name, 'content_type': 'video/mp4'}]}
+
+        if transcript_preferences:
+            request_data.update({'transcript_preferences': transcript_preferences})
+
+        bucket = Mock()
+        mock_conn.return_value = Mock(get_bucket=Mock(return_value=bucket))
+        mock_key_instance = Mock(
+            generate_url=Mock(
+                return_value='http://example.com/url_{file_name}'.format(file_name=file_name)
+            )
+        )
+        # If extra calls are made, return a dummy
+        mock_key.side_effect = [mock_key_instance] + [Mock()]
```

Comment:
It is hard to understand the above mock magic. Can we simplify it or add more comments?